src/components/controller.cpp

```diff
@@ -127,15 +118,13 @@ controller::~controller() {
   m_sig.detach(this);
 
   m_log.trace("controller: Stop modules");
-  for (auto&& block : m_modules) {
-    for (auto&& module : block.second) {
-      auto module_name = module->name();
-      auto cleanup_ms = time_util::measure([&module] {
-        module->stop();
-        module.reset();
-      });
-      m_log.info("Deconstruction of %s took %lu ms.", module_name, cleanup_ms);
-    }
+  for (auto&& module : m_modules) {
+    auto module_name = module->name();
+    auto cleanup_ms = time_util::measure([&module] {
+      module->stop();
+      module.reset();
```

Comment:
```suggestion
```
This line is now useless, since there are several instances of this `shared_ptr` it doesn't destroy anything.
Another solution is to add the following code before this loop but I don't think this is very useful since the module are deleted automatically when they objects are destroyed.
```cpp
m_blocks.clear();
```